pkg/autoops/api/api.go

```diff
@@ -251,95 +258,198 @@ func (s *AutoOpsService) validateCreateAutoOpsRuleRequest(
+			return statusInternal.Err()
+		}
+		return dt.Err()
 	}
-	if err := s.validateOpsEventRateClauses(req.Command.OpsEventRateClauses); err != nil {
+	if err := s.validateOpsEventRateClauses(req.Command.OpsEventRateClauses, localizer); err != nil {
 		return err
 	}
-	if err := s.validateDatetimeClauses(req.Command.DatetimeClauses); err != nil {
+	if err := s.validateDatetimeClauses(req.Command.DatetimeClauses, localizer); err != nil {
 		return err
 	}
-	if err := s.validateWebhookClauses(req.Command.WebhookClauses); err != nil {
+	if err := s.validateWebhookClauses(req.Command.WebhookClauses, localizer); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (s *AutoOpsService) validateOpsEventRateClauses(clauses []*autoopsproto.OpsEventRateClause) error {
+func (s *AutoOpsService) validateOpsEventRateClauses(
+	clauses []*autoopsproto.OpsEventRateClause,
+	localizer locale.Localizer,
+) error {
 	for _, c := range clauses {
-		if err := s.validateOpsEventRateClause(c); err != nil {
+		if err := s.validateOpsEventRateClause(c, localizer); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (s *AutoOpsService) validateOpsEventRateClause(clause *autoopsproto.OpsEventRateClause) error {
+func (s *AutoOpsService) validateOpsEventRateClause(
+	clause *autoopsproto.OpsEventRateClause,
+	localizer locale.Localizer,
+) error {
 	if clause.VariationId == "" {
-		return localizedError(statusOpsEventRateClauseVariationIDRequired, locale.JaJP)
+		dt, err := statusOpsEventRateClauseVariationIDRequired.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "variation_id"),
+		})
+		if err != nil {
+			return statusInternal.Err()
+		}
+		return dt.Err()
 	}
 	if clause.GoalId == "" {
-		return localizedError(statusOpsEventRateClauseGoalIDRequired, locale.JaJP)
+		dt, err := statusOpsEventRateClauseGoalIDRequired.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "goal_id"),
+		})
+		if err != nil {
+			return statusInternal.Err()
+		}
+		return dt.Err()
 	}
 	if clause.MinCount <= 0 {
-		return localizedError(statusOpsEventRateClauseMinCountRequired, locale.JaJP)
+		dt, err := statusOpsEventRateClauseMinCountRequired.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "min_count"),
+		})
+		if err != nil {
+			return statusInternal.Err()
+		}
+		return dt.Err()
 	}
 	if clause.ThreadsholdRate > 1 || clause.ThreadsholdRate <= 0 {
-		return localizedError(statusOpsEventRateClauseInvalidThredshold, locale.JaJP)
+		dt, err := statusOpsEventRateClauseInvalidThredshold.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "thredshold"),
```

Comment:
```suggestion
			Message: localizer.MustLocalizeWithTemplate(locale.InvalidArgumentError, "threshold"),
```

Reply:
@masaaania 
=

Reply:
`thredshold` is typo.

In fact, `thredshold` has been used elsewhere in the code, but I thought it should be fixed since this part is visible to the user.